pack.go

```diff
@@ -178,6 +229,71 @@ func packImage(ctx context.Context, pusher content.Pusher, configMediaType strin
 	return manifestDesc, nil
 }
 
+// packImage packs the given blobs, generates an image manifest for the pack,
+// and pushes it to a content storage.
+// If succeeded, returns a descriptor of the manifest.
+// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/manifest.md#guidelines-for-artifact-usage
+func packImage(ctx context.Context, pusher content.Pusher, artifactType string, layers []ocispec.Descriptor, opts PackOptions) (ocispec.Descriptor, error) {
+	var configDesc ocispec.Descriptor
+	if opts.ConfigDescriptor != nil {
+		configDesc = *opts.ConfigDescriptor
+	} else {
+		// use the empty descriptor for config
+		configDesc = ocispec.DescriptorEmptyJSON
+		configDesc.Annotations = opts.ConfigAnnotations
+		configData := ocispec.DescriptorEmptyJSON.Data
+		if err := pusher.Push(ctx, configDesc, bytes.NewReader(configData)); err != nil && !errors.Is(err, errdef.ErrAlreadyExists) {
+			return ocispec.Descriptor{}, fmt.Errorf("failed to push config: %w", err)
+		}
```

Comment:
If the `pusher` is a registry/repository/blob store, it will send out two requests to the remote.

Before `push`, we can assert if `pusher` can do `Exists()`. Since the empty blob is likely to exist, it will save one request on average.

Reply:
Good point!